src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,435 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Dynamo.Core;
+using Dynamo.Engine;
+using Dynamo.Graph.Nodes;
+using Dynamo.Library;
+using Dynamo.Models;
+using Dynamo.Search;
+using Dynamo.Search.SearchElements;
+using Dynamo.Utilities;
+using ProtoCore.AST.AssociativeAST;
+using ProtoFFI.Reflection;
+using ProtoCore.Utils;
+using ProtoFFI;
+using static Dynamo.Engine.LibraryServices;
+
+namespace NodeDocumentationMarkdownGenerator
+{
+    internal static class AssemblyHandler
+    {
+        /// <summary>
+        /// Scans a list of assemblies using a Reflection only load context to determine which nodes are in the assemblies.
+        /// </summary>
+        /// <param name="assemblyPaths">List of dll paths that should be scanned</param>
+        /// <param name="additionalPathsToLoad">List of dll paths that should be added to the PathAssemblyResolver. This can be used when there are types in the assembly that depends on coming from an external assembly that are not part of the assembly paths</param>
+        /// <returns></returns>
+        internal static List<MdFileInfo> ScanAssemblies(IEnumerable<string> assemblyPaths, IEnumerable<string> additionalPathsToLoad = null)
+        {
+            var paths = GetDefaultPaths();
+            paths.AddRange(assemblyPaths);
+            if (additionalPathsToLoad != null)
+            {
+                paths.AddRange(additionalPathsToLoad);
+            }
+
+            var resolver = new PathAssemblyResolver(paths);
+            var mlc = new MetadataLoadContext(resolver);
+
+            return Scan(assemblyPaths, mlc, resolver);
+        }
+
+        private static List<MdFileInfo> Scan(IEnumerable<string> assemblyPaths, MetadataLoadContext mlc, PathAssemblyResolver resolver)
+        {
+            var fileInfos = new List<MdFileInfo>();
+            var nodeSearchModel = new NodeSearchModel();
+            var pathManager = new PathManager(new PathManagerParams());
+
+            Console.WriteLine($"Starting scan of following assemblies: {string.Join(", ", assemblyPaths)}");
+
+            using (mlc)
+            {
+                var dynamoCoreAss = resolver.Resolve(mlc, new AssemblyName("DynamoCore"));
+                var nodeModelType = dynamoCoreAss.GetType("Dynamo.Graph.Nodes.NodeModel");
```

Comment:
not necessarily trying to avoid a dependency on DynamoCore, it's more that I need to make sure that I get the correct reference to the NodeModel type. The issue I'm coming across here is that `typeof(NodeModel)` is pointing to a different reference than the nodes we are scanning as they are only loaded in reflection. 

So if i try to use the original 
```csharp
internal static bool IsNodeSubType(Type t)
{
    return //t.Namespace == "Dynamo.Graph.Nodes" &&
        !t.IsAbstract &&
            t.IsSubclassOf(typeof(NodeModel))
            && t.GetConstructor(Type.EmptyTypes) != null;
}
```
it will never return true, even when the type is subclass of NodeModel as `t` is reflection loaded whereas `typeof(NodeModel)` is not, and therefore they will never be equal when compared.

Reply:
understood, thanks